plugins/aiops/pkg/gateway/metricai.go

```diff
@@ -0,0 +1,363 @@
+	}
+	return &metricai.MetricAIRunJobResponse{JobRunId: res["JobRunId"].(string), SubmittedTime: time.Now().String(), Status: res["Status"].(string)}, nil
+}
+
+func (p *AIOpsPlugin) CreateJob(ctx context.Context, jobRequest *metricai.MetricAICreateJobRequest) (*metricai.MetricAIAPIResponse, error) {
+	// use the info provided by user to create a job
+	// Info includes: job's name, the cluseter_id, a list of namespaces to watch.
+	ctxca, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+	metricAIKeyValue, err := p.metricAIJobKv.GetContext(ctxca)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to ListJobRuns for metricAI: %v", err)
+	}
+	jid := strings.ToLower(jobRequest.JobId)
+
+	if jid == "" { // id can't be empty
+		return nil, &RequestError{
+			StatusCode: 503,
+			Err:        errors.New("jobId can't be empty"),
+		}
+	}
+
+	if strings.Contains(jid, jobRunDelimiter) { // disallow the delimiter. TODO: should only allow chars include alphanum and - and _
+		return nil, &RequestError{
+			StatusCode: 503,
+			Err:        errors.New("jobId can't contain special char " + jobRunDelimiter),
+		}
+	}
+
+	if _, err := metricAIKeyValue.Get(jid); err == nil { // check if this id exists
+		return nil, &RequestError{
+			StatusCode: 503,
+			Err:        errors.New("The jobId to add already exist"),
+		}
+	}
+
+	job := make(map[string]interface{})
+	job["JobId"] = jid
+	job["JobCreateTime"] = time.Now().String()
+	job["ClusterId"] = jobRequest.ClusterId
+	job["Namespaces"] = jobRequest.Namespaces
+	job["JobDescription"] = jobRequest.JobDescription
+	jobJsonStr, err := json.Marshal(job)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to marshal JobStatus for metricAI: %v", err)
+	}
+	_, err = metricAIKeyValue.Put(jid, []byte(jobJsonStr))
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to CreateJob for metricAI: %v", err)
+	}
+	return &metricai.MetricAIAPIResponse{SubmittedTime: time.Now().String(), Status: "Success"}, nil
+}
+
+// delete job_id from the natsKV bucket. This won't delete the job_run attached to this job_id
+func (p *AIOpsPlugin) DeleteJob(ctx context.Context, jobId *metricai.MetricAIId) (*metricai.MetricAIAPIResponse, error) {
+	ctxca, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+	metricAIKeyValue, err := p.metricAIJobKv.GetContext(ctxca)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to DeleteJob for metricAI: %v", err)
+	}
+	jid := jobId.Id
+	if _, err := metricAIKeyValue.Get(jid); err != nil {
+		if errors.Is(err, nats.ErrKeyNotFound) {
+			return nil, &RequestError{
+				StatusCode: 503,
+				Err:        errors.New("The jobId to delete doesn't exist"),
+			}
+		}
+		return nil, status.Errorf(codes.Internal, "Failed to DeleteJob for metricAI: %v", err)
+	}
+	metricAIKeyValue.Delete(jid)
+	return &metricai.MetricAIAPIResponse{Status: "Success", Description: "The JobId key :" + jid + " is deleted"}, nil
```

Comment:
Nit: rather than using concatenation we should use sprintf e.g
```go
Description: fmt.Sprintf("The JobId key :%s is deleted", jid)
```

Reply:
fixed all the similar messages